tests/utils.py

```diff
@@ -0,0 +1,40 @@
+import sysconfig
+from pathlib import Path
+
+
+def use_pyximport(libname: str) -> bool:
+    """Attempts to import the cython test library, and returns a boolean indicating success."""
+    so_ext = sysconfig.get_config_var("EXT_SUFFIX")
+    so = Path("tests") / f"_test_{libname}{so_ext}"
+    paths = [
+        Path("tests") / f"_test_{libname}.c",
+        Path("tests") / f"_test_{libname}.pyxbldc",
+        so,
+    ]
+    for path in paths:
+        path.unlink(missing_ok=True)
+
+    try:
+        import ctypes
+        import importlib
+
+        import pyximport
+
+        pyximport.install(build_in_temp=False, inplace=True)
+        importlib.import_module(f"_test_{libname}")
+        ctypes.cdll.LoadLibrary(f"{so}")
+    except ImportError as ex:
+        patterns = [
+            "distutils.compilers.C.errors.CompileError",
+            "Cython.Compiler.Errors.CompileError",
+        ]
+        for pattern in patterns:
+            if pattern in f"{ex}":
+                raise Exception(f"Cannot compile {libname}:\n\t{ex}") from ex
```

Comment:
Again for my own edification - why propagate these specific exceptions instead of returning `False`? 

It might be helpful to include `Returns` and `Raises` blocks in the docstring to explain the circumstances when this function doesn't catch and return `False`

Reply:
Since if `pyximport` isn't found, the tests are skipped, but if the cython code can't compile, then that's an exception.